x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * A connector document consists of:
+ * <ul>
+ *     <li>...</li>
+ * </ul>
+ */
+public class Connector implements Writeable, ToXContentObject {
+
+    private final String connectorId;
+    @Nullable
+    private final String apiKeyId;
+
+    private final Map<String, Object> configuration;
+
+    private final ConnectorCustomSchedule customScheduling;
```

Comment:
If I understand the [custom scheduling docs](https://www.elastic.co/guide/en/enterprise-search/8.9/crawler-managing.html#crawler-managing-multiple-schedules-creating-multiple-schedules) correctly, shouldn't this be a `Map<String, ConnectorCustomSchedule>`, so that you can have multiple custom schedules?

Reply:
Yes! That's a good spot 👍 I realised it while adding unit tests and just pushed a fix. Thank you for spotting bugs ;)